resources/build/test/build-bot/trigger-build-bot.test.sh

```diff
@@ -63,10 +65,10 @@ _do_test_build_bot_check_messages_file() {
   build_bot_check_messages $prnum "$(cat ${THIS_SCRIPT_PATH}/pr-$prnum-data.txt)" "$(cat ${THIS_SCRIPT_PATH}/pr-$prnum-commits.txt)"
 
   for i in "${!expected_build_platforms[@]}"; do
-    assert-equal "${build_platforms[$i]}" "${expected_build_platforms[$i]}" "PR #$prnum: build_platforms[$i]"
+    assert-equal "${build_platforms[$i]:-missing}" "${expected_build_platforms[$i]:-missing}" "PR #$prnum: build_platforms[$i]"
```

Comment:
Where does `-missing` coming from?

Reply:
This is so the script fails with a legible assertion failure if the `$build_platforms` array is missing a value at index `$i`, rather than with an (obscure) script error. I experienced this when I was running tests in developing this. The `missing` value does not appear on successful runs. 

I guess the `$expected_build_platorms[$i]` should never be missing because this is a loop iterating over that array!